hazelcast/include/hazelcast/client/impl/metrics/metrics_dictionary.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "hazelcast/util/byte.h"
+#include "hazelcast/util/export.h"
+
+namespace hazelcast {
+namespace client {
+namespace impl {
+namespace metrics {
+
+class HAZELCAST_API metrics_dictionary
+{
+private:
+    using map = std::map<std::string, int>;
+public:
+    using const_iterator = map::const_iterator;
+
+    int get_dictionary_id(const std::string &word);
+
+    const_iterator begin() const noexcept;
```

Comment:
these are for testing only, right? you can either clearly document this or make it private and use friend.

Reply:
No. They are for this: 

> test `metrics_compressor_test, test_blob` fails on my local. Output:
> 
> ```
> /Users/ihsan/Desktop/work/src/hazelcast-cpp-client/hazelcast/test/src/metrics_test.cpp:108: Failure
> Expected equality of these values:
>   expected_blob
>     Which is: { '\x1' (1), '\0', '\0', '\0', '\x98' (152), 'x' (120, 0x78), '\x1' (1), '\xED' (237), '\x94' (148), 'Y' (89, 0x59), '\xE' (14), '\xC2' (194), '0' (48, 0x30), '\f' (12, 0xC), '\x5' (5), '\x87' (135), 'P' (80, 0x50), '(' (40, 0x28), '\xFB' (251), 'z' (122, 0x7A), '\x12' (18), '\xB6' (182), '\x3' (3), 'A' (65, 0x41), '\x1' (1), '\xB5' (181), '\b' (8), '\n' (10, 0xA), '\x2' (2), '\x84' (132), '\xB8' (184), '=' (61, 0x3D), ... }
>   actual_blob
>     Which is: { '\0', '\x1' (1), '\0', '\0', '\0', '\x98' (152), 'x' (120, 0x78), '\x1' (1), '\xED' (237), '\x94' (148), 'Y' (89, 0x59), '\xE' (14), '\xC2' (194), '0' (48, 0x30), '\f' (12, 0xC), '\x5' (5), '\x87' (135), 'P' (80, 0x50), '(' (40, 0x28), '\xFB' (251), 'z' (122, 0x7A), '\x12' (18), '\xB6' (182), '\x3' (3), 'A' (65, 0x41), '\x1' (1), '\xB5' (181), '\b' (8), '\n' (10, 0xA), '\x2' (2), '\x84' (132), '\xB8' (184), ... }
> With diff:
> @@ -1,2 +1,2 @@
> -{ '\x1' (1), '\0', '\0', '\0', '\x98' (152), 'x' (120, 0x78), '\x1' (1), '\xED' (237), '\x94' (148), 'Y' (89, 0x59), '\xE' (14), '\xC2' (194), '0' (48, 0x30), '\f' (12, 0xC), '\x5' (5), '\x87' (135), 'P' (80, 0x50), '(' (40, 0x28), '\xFB' (251), 'z' (122, 0x7A), '\x12' (18), '\xB6' (182), '\x3' (3), 'A' (65, 0x41), '\x1' (1), '\xB5' (181), '\b' (8), '
> -' (10, 0xA), '\x2' (2), '\x84' (132), '\xB8' (184), '=' (61, 0x3D), ... }
> +{ '\0', '\x1' (1), '\0', '\0', '\0', '\x98' (152), 'x' (120, 0x78), '\x1' (1), '\xED' (237), '\x94' (148), 'Y' (89, 0x59), '\xE' (14), '\xC2' (194), '0' (48, 0x30), '\f' (12, 0xC), '\x5' (5), '\x87' (135), 'P' (80, 0x50), '(' (40, 0x28), '\xFB' (251), 'z' (122, 0x7A), '\x12' (18), '\xB6' (182), '\x3' (3), 'A' (65, 0x41), '\x1' (1), '\xB5' (181), '\b' (8), '
> +' (10, 0xA), '\x2' (2), '\x84' (132), '\xB8' (184), ... }
> ```

Probably because Apple has its own Zlib implementation and is producing a different binary. This won't matter if we use our implementation.

Reply:
begin() and end() calls are used by the for loop in metrics_compressor::write_dictionary.